sys/phydat/phydat_str.c

```diff
@@ -78,14 +80,35 @@ const char *phydat_unit_to_str(uint8_t unit)
     }
 }
 
-char phydat_scale_to_str(int8_t scale)
+char phydat_scale_to_str(int8_t scale, uint8_t unit)
 {
+    /* extra handling for multi and non dimensional units */
+    switch (unit) {
+        case UNIT_M2:       /* rescale for 2d unit */
+            if (scale % 2 != 0) {
```

Comment:
why? IMO the scale allows for enhanced accuracy without using floats, so one could have scale = +/-5 for square or cubic meters, as well - or not?